map/styled_point.cpp

```diff
@@ -0,0 +1,51 @@
+#include "map/styled_point.hpp"
+
+graphics::DisplayList * StyledPoint::GetDisplayList(UserMarkDLCache * cache) const
+{
+  UserMarkContainer const * container = GetContainer();
+  UserMarkDLCache::Key const key =
+      GetStyle().empty()
+          ? container->GetDefaultKey()
+          : UserMarkDLCache::Key(GetStyle(), graphics::EPosAbove, container->GetDepth());
+  return cache->FindUserMark(key);
+}
+
+double StyledPoint::GetAnimScaleFactor() const
+{
+  // Matches the behaviour for non-custom drawables. The only caller
+  // of ::DrawUserMark is UserMarkContainer::Draw and it always passes
+  // this value.
+  return 1.0;
+}
+
+m2::PointD const & StyledPoint::GetPixelOffset() const
+{
+  static m2::PointD const s_centre(0.0, 0.0);
+  static m2::PointD const s_offset(0.0, 3.0);
+
+  return GetStyle().empty() ? s_centre : s_offset;
+}
+
+static char const * s_arrSupportedColors[] = {
```

Comment:
Please, don't use this archaic static global variables, use anonymous namespaces instead. Also
in accordance with our style guides this variable should be named kArrSupportedColors.